### PR TITLE
Generate avatars with author initials as fallback of gravatar

### PR DESCRIPTION
**Pull request by pmiossec**
A smaller step than #6417 as it requires no ("not everyone agrees") UI changes but bringing most of the added value...

## Proposed changes

- Add a new new gravatar fallback possibility with avatars generated from author initials

### Before
image

image

### After

image

image

## Test methodology 

* Unit tests
* Manual

## Test environment(s) 

- Git Extensions 3.1.0
- Build c937777048244969bee4b89f19e641543dd9a6e7
- Git 2.20.1.windows.1
- Microsoft Windows NT 10.0.17134.0
- .NET Framework 4.7.3362.0
- DPI 96dpi (no scaling)

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/Avatars/InitialsAvatarGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            }
+
+            return (initials, hashCode);
+        }
+
+        private readonly Graphics _graphics = Graphics.FromImage(new Bitmap(1, 1));
+        private readonly Brush _textBrush = new SolidBrush(Color.WhiteSmoke);
+        private readonly Color[] _avatarColors =
+        {
+            Color.RoyalBlue,
```
Comment (gerhardol) [inline GitUI/Avatars/InitialsAvatarGenerator.cs]: Note for theming: May need customization. (I thought about asking for more colors too, but it should be OK).

File: GitUI/Avatars/InitialsAvatarGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+                drawing.SmoothingMode = SmoothingMode.AntiAlias;
+                drawing.TextRenderingHint = TextRenderingHint.ClearTypeGridFit;
+                var x = textSize.Width >= textSize.Height ? 0 : (textSize.Height - textSize.Width) / 2;
+                var y = textSize.Width >= textSize.Height ? (textSize.Width - textSize.Height) / 2 : 0;
+                drawing.DrawString(text, font, _textBrush, x, y);
+
+                drawing.Save();
+
+                drawing.Dispose();
```
Comment (RussKie) [inline GitUI/Avatars/InitialsAvatarGenerator.cs]: Could we instead write
```
using (var drawing = Graphics.FromImage(img))
{
    drawing.Clear(backColor);
    // ...
    drawing.Save();
}
```

Comment (pmiossec) [reply GitUI/Avatars/InitialsAvatarGenerator.cs]: Indeed. I don't know why I haven't seen it.
Anyway... fixed.

**Comment by RussKie** [comment]
Works well 👍

**Pull request merged**